Added possibility to bind ng-drag-scroll to DOM elements

Shocoben opened the pull request:
Hi, now when you add the direct ng-drag-scroll. You can add an attribute
"scroll-element"  which contains a jQlite selector. I have also
refactorised the HitTest function.

I know that you're the main fork of this project. But that main project hasn't already accepted your pull request. So I've forked your modification for the touch devices and added the functionality with it and tested it on touch devices. And it works ! :)

Shocoben committed:
Added possibility to bind ng-drag-scroll to DOM elements

When you add the direct ng-drag-scroll. You cann add an attribute
"scroll-element"  which contains a jQlite selector. I have also
refactorised the HitTest function.

Shocoben committed:
Improved the ngDragClone

- Now, when being hover a ngDroppable, the ngDragClone also gains the
class "drag-over"

- Attribute ng-drag-clone-group="groupename" let your specify a group to
link a specific draggable to a specific ngDragClone

- Now, the ngDragClone automatically copy the html and classes of the
draggable.

- You can deactivate the copy with the attributes on both, the ngDrag
and NgDragClone by using the attribute ng-drag-clone-copy-class="false"
or ng-drag-clone-copy-html="false"

- Can add class to the clone by the ngDrag with the attribute
ng-drag-clone-add-class="test"

You can look at the "example-multiple-scrolls.html" for more details.

Shocoben commented:
I've also improved the ngDragClone element
- Now, when being hover a ngDroppable, the ngDragClone also gains the
  class "drag-over"
- Attribute ng-drag-clone-group="groupename" let your specify a group to
  link a specific draggable to a specific ngDragClone
- Now, the ngDragClone automatically copy the html and classes of the
  draggable.
- You can deactivate the copy with the attributes on both, the ngDrag
  and NgDragClone by using the attribute ng-drag-clone-copy-class="false"
  or ng-drag-clone-copy-html="false"
- Can add class to the clone by the ngDrag with the attribute
  ng-drag-clone-add-class="test"

You can look at the "example-multiple-scrolls.html" for more details.

bastbnl commented:
Hi,

Thanks for the improvements. The original author has not responded to my pr yet, so I'll take a closer look at the changes you made and merge them in. I'll get back to you on that 

Out of curiosity: "And it works !". Does that mean the initial fix works for your setup too?

Shocoben commented:
Hi, 

Yeah I've seen that :(

Yep, I wanted to say "Your fix worked on my samsung galaxy s4. And my feature didn't broke your fix because it's still working ! hehe".

Shocoben committed:
dragEnter and dragLeave event + ngDragClone can select in which DOM elment the copy is done

- dragEnter and dragLeave event on the drop zone
- ngDragClone can select in which DOM elment the copy is done. Example :
ng-drag-clone-copy-html=".clone_container"

Shocoben commented:
Hi, I've improved the ngDragClone copy function + Now you can add a callback to dragEnter and dragLeave on the drop zone.

Shocoben committed:
Debugged ngDragCloneHide and added doFollowMouse option

You can now ask the clone to do not hide the original drag, and you can
ask the original drag to not follow the mouse
